app/routes/questionnaire.py

```diff
@@ -283,6 +300,52 @@ def get_thank_you(schema):
     )
 
 
+@post_submission_blueprint.route("confirmation-email/send", methods=["GET", "POST"])
+@login_required
+def get_confirmation_email():
+    if not get_session_store().session_data.confirmation_email_sent:
+        raise NotFound
+
+    confirmation_email = ConfirmationEmail()
+
+    if request.method == "POST":
+        confirmation_email.handle_post()
+
+        if confirmation_email.form_valid:
+            return redirect(
+                url_for(
+                    "post_submission.get_confirmation_email_sent",
+                    email=confirmation_email.get_url_safe_serialized_email(),
+                )
+            )
+
+    return render_template(
+        template="confirmation-email",
+        content=confirmation_email.get_context(),
+        hide_signout_button=True,
+    )
+
+
+@post_submission_blueprint.route("confirmation-email/sent", methods=["GET"])
+@login_required
+def get_confirmation_email_sent():
+    if not get_session_store().session_data.confirmation_email_sent:
+        raise NotFound
+
+    email = URLParamSerializer().loads(request.args.get("email"))
+
+    return render_template(
+        template="confirmation-email-sent",
+        content={
+            "email": email,
+            "url": url_for("post_submission.get_confirmation_email"),
```

Comment:
The name `url` is ambiguous. Something like `send_confirmation_email_url` would be clearer.